docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md

```diff
@@ -24,3 +24,67 @@ This model consists of a series of modules, each producing a table which serves
 - User Mapping: Provides a mapping between user identifiers, `domain_userid` and `user_id`, outputting the table `snowplow_web_user_mapping`. This can be used for session stitching.
 
+## Optional Modules
+
+### Consent Tracking Custom Module
+
+
+### Introduction
+
+This custom module is built as an extension of the [dbt-snowplow-web package](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md) that transforms raw `consent_preferences` and `cmp_visible` event data into derived tables for easier querying generated by the [Enhanced Consent plugin](/docs/collecting-data/collecting-from-own-applications/javascript-trackers/javascript-tracker/javascript-tracker-v3/plugins/consent/index.md) of the [JavaScript tracker](/docs/collecting-data/collecting-from-own-applications/javascript-trackers/index.md)
+
+For the incremental logic to work within the module, one **prerequisite** is to use `RDB Loader v4.0.0` and above as the custom module relies on the additional `load_tstamp` field for dbt native incrementalisation.
+
+In order to keep the documentations separate and less verbose, this guide will assume the reader is already familiar with configuring and running the web model and will only explain what the custom consent module does and how to operate it in conjunction with the web model. Please refer to the [snowplow-web dbt docs](/docs/modeling-your-data/modeling-your-data-with-dbt/index.md) for a full breakdown of the package and how to set it up.
+
+
+### Overview
+
+This custom module consists of a series of dbt models with the goal to produce the following aggregated models from the raw consent tracking events:
+
+  - `snowplow_web_consent_log`: Snowplow incremental table showing the audit trail of consent and Consent Management Platform (cmp) events
+
+  - `snowplow_web_consent_users`: Incremental table of user consent tracking stats
+
+  - `snowplow_web_consent_totals`: Summary of the latest consent status as per consent version
+
+  - `snowplow_web_consent_scope_status`: Aggregate of current number of users consented to each consent scope
+
+  - `snowplow_web_cmp_stats`: Used for modeling cmp_visible events and related metrics
+
+  - `snowplow_web_consent_versions`: Incremental table used to keep track of each consent version and its validity
+
+
+### Operation
+
+It is assumed that the dbt_snowplow_web package is already installed and configured as per the [Quick Start](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-quickstart/index.md) instructions.
+
+#### Enable the module
+
+To enable the custom module simply copy the following code snippet to your own dbt_project.yml file:
+
+```yml
+# dbt_project.yml
+
+models:
+  snowplow_web:
+    optional_modules:
+      consent:
+        enabled: true
+        bigquery:
+          enabled: "{{ target.type == 'bigquery' | as_bool() }}"
+        snowflake:
+          enabled: "{{ target.type == 'snowflake' | as_bool() }}"
+```
```

Comment:
💡 Can we maybe call out somewhere this is currently only available for bigquery and snowflake?